CopilotKit/packages/shared/src/utils/parse-chat-completion.ts

```diff
@@ -1,5 +1,23 @@
 import { Role } from "../types/openai-assistant";
 
+export interface ToolCallFunctionCall {
+  arguments?: string;
+
+  name?: string;
+  // TODO:
+  // Temporarily add scope to the OpenAI protocol until we
+  // have our own protocol.
+  // When scope is "server", the client will not attempt to
+  // execute the function.
+  scope?: "client" | "server";
+}
+
+export interface ToolCallPayload {
+  index: number;
+  id?: string;
+  function: ToolCallFunctionCall;
+}
+
```

Comment:
these types are used in the associated value types of the 2 `Mode` types.
So we factor them out for convenience